dependencies update + more test codes + sample middleware code

pilinux opened the pull request:
## What's Changed

- added test files for gorm package https://github.com/onrik/logrus/commit/560b822c6e1805e07fd01e3ce0d4311330aac176
- added example integration with test codes https://github.com/onrik/logrus/commit/560b822c6e1805e07fd01e3ce0d4311330aac176
- bumped dependencies https://github.com/onrik/logrus/commit/2c50815c62783464dc9980806017eaef20f235d4

pilinux committed:
ci: added test files

pilinux committed:
bumped dependencies

pilinux committed:
bump golang.org/x/term from 0.14.0 to 0.15.0

pilinux committed:
`onrik/logrus` -> `pilinux/logrus`

onrik commented on echo/README.md:
```diff
@@ -6,7 +6,7 @@ package main
 import (
     "github.com/labstack/echo/v4"
-    echolog "github.com/onrik/logrus/echo"
+    echolog "github.com/pilinux/logrus/echo"
```
Looks like it's redundant.

pilinux committed:
bumped dependencies

github.com/getsentry/sentry-go v0.25.0 -> v0.27.0
github.com/labstack/echo/v4 v4.11.3 -> v4.11.4
github.com/labstack/gommon v0.4.1 -> v0.4.2
github.com/stretchr/testify v1.8.4 -> v1.9.0
github.com/tidwall/gjson v1.17.0 -> v1.17.1
golang.org/x/term v0.15.0 -> v0.17.0

pilinux committed:
ci: github actions

pilinux committed:
ci: explicitly set go v1.20.14

https://github.com/actions/setup-go/issues/328

pilinux committed:
ci: upload coverage report during `push` event

pilinux committed:
chore: bumped dependencies

pilinux committed:
chore: bumped dependencies

pilinux committed:
ci: bump golangci-lint version

dependabot[bot] committed:
Bump github.com/getsentry/sentry-go from 0.31.1 to 0.37.0

Bumps [github.com/getsentry/sentry-go](https://github.com/getsentry/sentry-go) from 0.31.1 to 0.37.0.
- [Release notes](https://github.com/getsentry/sentry-go/releases)
- [Changelog](https://github.com/getsentry/sentry-go/blob/master/CHANGELOG.md)
- [Commits](https://github.com/getsentry/sentry-go/compare/v0.31.1...v0.37.0)

---
updated-dependencies:
- dependency-name: github.com/getsentry/sentry-go
  dependency-version: 0.37.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

pilinux committed:
Merge pull request #31 from pilinux/dependabot/go_modules/github.com/getsentry/sentry-go-0.37.0

Bump github.com/getsentry/sentry-go from 0.31.1 to 0.37.0
